pkg/apps/metrics/prometheus/metrics.go

```diff
@@ -0,0 +1,100 @@
+package prometheus
+
+import (
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"k8s.io/apimachinery/pkg/labels"
+
+	internalversion "github.com/openshift/origin/pkg/apps/generated/listers/apps/internalversion"
+	"github.com/openshift/origin/pkg/apps/util"
+)
+
+const (
+	completeRolloutCount = "complete_rollouts_total"
+	activeRolloutCount   = "active_rollouts_total"
+
+	availablePhase = "available"
+	failedPhase    = "failed"
+	cancelledPhase = "cancelled"
+)
+
+var (
+	nameToQuery = func(name string) string {
+		return strings.Join([]string{"openshift_apps", name}, "_")
+	}
+
+	completeRolloutCountDesc = prometheus.NewDesc(
+		nameToQuery(completeRolloutCount),
+		"Counts total complete rollouts",
+		[]string{"phase"}, nil,
+	)
+
+	activeRolloutCountDesc = prometheus.NewDesc(
+		nameToQuery(activeRolloutCount),
+		"Counts total active rollouts",
+		[]string{"namespace", "name", "phase"}, nil,
+	)
+
+	apps       = appsCollector{}
+	registered = false
+)
+
+type appsCollector struct {
+	lister internalversion.DeploymentConfigLister
+}
+
+func IntializeMetricsCollector(dcLister internalversion.DeploymentConfigLister) {
+	apps.lister = dcLister
+	if !registered {
+		prometheus.MustRegister(&apps)
+		registered = true
+	}
+	glog.V(4).Info("apps metrics registered with prometheus")
+}
+
+func (c *appsCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- completeRolloutCountDesc
+	ch <- activeRolloutCountDesc
+}
+
+// Collect implements the prometheus.Collector interface.
+func (c *appsCollector) Collect(ch chan<- prometheus.Metric) {
+	result, err := c.lister.List(labels.Everything())
+	if err != nil {
+		glog.V(4).Infof("Collecting metrics for apps failed: %v", err)
+		return
+	}
+
+	var available, failed, cancelled float64
+
+	for _, d := range result {
+		if util.IsTerminatedDeployment(d) {
+			if util.IsDeploymentCancelled(d) {
+				cancelled++
+				continue
+			}
+			if util.IsFailedDeployment(d) {
+				failed++
+				continue
+			}
+			if util.IsCompleteDeployment(d) {
+				available++
+				continue
+			}
+		}
+
+		generation := d.Status.ObservedGeneration
+		ch <- prometheus.MustNewConstMetric(activeRolloutCountDesc, prometheus.GaugeValue, float64(generation), []string{
```

Comment:
@smarterclayton i'm not sure if generation is the right value here, the builds are using start timestamp (unix)